tensorflow/core/kernels/data/dataset_test_base.cc

```diff
@@ -560,5 +560,203 @@ Status DatasetOpsTestBase::AddDatasetInput(
   return Status::OK();
 }
 
+Status DatasetOpsTestBase::EvaluateDatasetNodeName(
```

Comment:
I think these test methods should be renamed from `Evaluate*` to `Check*` and use `EXPECT_*` internally to check expectations and only return non-ok status on error (instead of returning errors if an expectation fails).

For instance, this method would look like this:

 ```
Status DatasetOpsTestBase::CheckDatasetNodeName(
    const DatasetBase& dataset, const string& expected_dataset_node_name) {
  EXPECT_EQ(dataset.node_name(), expected_dataset_node_name);
  return Status::OK();
}
```

and the caller should use it as `TF_ASSERT_OK(CheckDatasetNodeName(dataset, expected));`